Python/thread_pthread.h

```diff
@@ -396,6 +404,10 @@ PyThread_get_thread_native_id(void)
 #elif defined(__DragonFly__)
     lwpid_t native_id;
     native_id = lwp_gettid();
+#elif defined(__GNU__)
+    mach_port_t native_id;
+    native_id = mach_thread_self();
+    mach_port_deallocate(mach_task_self(), mach_thread_self());
```

Comment:
It's not obvious to me what the semantics of this are. Is the return value of `mach_thread_self` globally unique accross the system? Is it the same value when called several times from a given thread?

Reply:
It is unique within the process that calls mach_thread_self. It is always the same when calling again, because glibc keeps a reference to it while the thread is running.

Reply:
The doc mandates that the native id is unique system-wide:
https://docs.python.org/3/library/_thread.html#thread.get_native_id

Reply:
ah, then GNU/Hurd doesn't have such a thing for threads

Reply:
I have kept only the kfreebsd part then.